extensions/shared/external-media/external-media-modal.js

```diff
@@ -0,0 +1,288 @@
+import apiFetch from '@wordpress/api-fetch';
+import { compose } from '@wordpress/compose';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { Component } from '@wordpress/element';
+import { withNotices, Modal } from '@wordpress/components';
+import { UP, DOWN, LEFT, RIGHT } from '@wordpress/keycodes';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { PATH_RECENT } from './constants';
+import MediaItem from './media-browser/media-item';
+
+const MODAL_NAME = 'edit-post/external-media';
+
+const CopyingMedia = ( { items } ) => {
+	const classname =
+		items.length === 1
+			? 'jetpack-external-media-browser__single'
+			: 'jetpack-external-media-browser';
+
+	return (
+		<div className={ classname }>
+			<div className="jetpack-external-media-browser__media">
+				{ items.map( item => (
+					<MediaItem item={ item } key={ item.ID } isSelected isCopying />
+				) ) }
+			</div>
+		</div>
+	);
+};
+
+class ExternalMediaModal extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			media: [],
+			nextHandle: false,
+			isLoading: false,
+			isCopying: null,
+			requiresAuth: false,
+			path: { ID: PATH_RECENT },
+		};
+
+		this.arrowKeysPropagationStopped = false;
+	}
+
+	componentDidUpdate() {
+		this.handleArrowKeysPropagation();
+	}
+
+	/**
+	 * When the External Media modal is open, pressing any arrow key causes it to
+	 * close immediately. This is happening because React fires blur event on the
+	 * Image block re-render, which seems to be triggered only by arrow keydown
+	 * event (possibly caught by the editor canvas during the re-render cycle).
+	 * Right after that, it tries to restore the focus to the modal element, but
+	 * because the blur has already been triggered, modal's "withFocusOutside"
+	 * handler thinks that the outside of the modal was clicked so it closes it
+	 * anyway. It wouldn't happen if the modal was isolated from the Image block
+	 * renderer, but because of the nature of this implementation (customized
+	 * media upload component) we're not really able to do that at this moment.
+	 *
+	 * This handler makes sure that the keydown event doesn't propagate further,
+	 * which fixes the issue described above while still keeping arrow keys
+	 * functional inside the modal (i.e. when navigating inside Pexels seach input
+	 * text or choosing filter from one of the Google Photos filters' select
+	 * dropdown).
+	 */
+	handleArrowKeysPropagation() {
+		const eventListenerEl = document.querySelector( '.components-modal__content' );
+
+		if ( eventListenerEl && ! this.arrowKeysPropagationStopped ) {
+			eventListenerEl.addEventListener( 'keydown', () => {
+				if ( [ UP, DOWN, LEFT, RIGHT ].includes( event.keyCode ) ) {
+					event.stopPropagation();
+				}
+			} );
+
+			this.arrowKeysPropagationStopped = true;
+		}
+
+		if ( ! eventListenerEl && this.arrowKeysPropagationStopped ) {
+			this.arrowKeysPropagationStopped = false;
+		}
+	}
```

Comment:
A modal should catch _all_ keypresses within it. Nothing should propagate outside, but an escape keypress should close the modal. So, rather than catch specific keys (arrows) and exclude them, should we really just be trapping _all_ keys?

Reply:
> A modal should catch all keypresses within it.

This would be true for our modal, but i.e. the `KeboardShortcutHelpModal` uses global key bindings for visibility toggle (`^⌥H` to open it from anywhere in the editor). So if all the keys were contained, that toggle would not work.

In our modal, escape is already handled and contained [here](https://github.com/WordPress/gutenberg/blob/30bc64c73daeae879f344992e9f74131d7366acb/packages/components/src/modal/frame.js#L84). If we stop the propagation of all keys, the [original  `keydown` handler of the modal](https://github.com/WordPress/gutenberg/blob/30bc64c73daeae879f344992e9f74131d7366acb/packages/components/src/modal/frame.js#L69) would not be reached, effectively disabling the escape key. We could instead block everything **except** escape. but I'm not sure if that's necessary? What if we block some global bindings too?